.github/workflows/go-test.yaml

```diff
@@ -14,7 +14,8 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v4
         with:
-          go-version: 1.20.x
+          go-version: '^1.20.3'
```

Comment:
the `.x` is valid semver
https://github.com/npm/node-semver#x-ranges-12x-1x-12-